### PR TITLE
[Elastic Agent] Golden files for program tests

### PR DESCRIPTION
**Pull request by ruflin**
Currently it seems the expected config files for each program were manually generated. This makes adding a new test cumbersome and in case of small changes it means adjusting many files.

This changes the test that it is possible to run with `go test . -generate` and new golden files are generated. Ideally it also means most of the additional tests become obsolete like how many program files should be generated. To make sure no left over files exist, on -generate all the generated files are first cleaned up.

NOTE: This PR is not complete yet, for example removal of old files is missing and cleanup of generated files. I wanted to discuss the approach first before going further.

### CODE REVIEW

**Comment by ruflin** [comment]
@blakerouse @michalpristas Would be great to get your feedback on this idea.

File: x-pack/elastic-agent/pkg/agent/program/program_test.go [hunk 1]
```diff
@@ -436,6 +442,16 @@ func TestConfiguration(t *testing.T) {
 		},
 	}
 
+	// Cleanup all generated files to make sure not having any left overs
+	if *generateFlag {
+		generatedFiles, err := filepath.Glob(filepath.Join("testdata", "*.generated.yml"))
```
Comment (ruflin) [inline x-pack/elastic-agent/pkg/agent/program/program_test.go:447]: Better name? Maybe .golden.yml?

**Review by blakerouse — CHANGES_REQUESTED** [review]
Some what confused on what you are changing here? It seems like you are changing the code to generated the expected results of the tests. To me this will negate the point of the tests. The tests are ensuring that the inputs, match the outputs. The outputs are hand written to ensure that the output is what is expected with the given input.

**Comment by ruflin** [comment]
The test files are only generated if the `-generate` flag is used. Otherwise I agree it would defy the purpose of the tests completely. This is known under [golden files test](https://medium.com/@jarifibrahim/golden-files-why-you-should-use-them-47087ec994bf) (I picked the first Google result ;-) ).

The problem with the current approach is that it is really cumbersome to add new examples. With the new approach, the first time the generate script can be run and the output can be reviewed. If in the future there is a "mass change" needed, this also make it easier. The important part is that any time the `-generate` flag is used, the diffs of the files must be carefully reviewed to make sure it is part of the expected change. But that is what a PR review is for.

**Comment by ruflin** [comment]
To make this PR easier to review I did not change the old tests and kept them as is. I think we can remove some of these as the new ones replace part of the old ones that have at least one program but we can do this in a follow up PR.

Marking this as ready for review.

**Review by blakerouse — APPROVED** [review]
Thanks for keeping the old tests. I think its good so we don't cause any regressions, until we know for sure this will cover those old tests.

**Pull request merged**